pull in gtest source into subdir gtestsrc

Zhziyao opened the pull request:
replace '--enable-gtest=...' with '--enable-gtest' in the cmd of compiling libyami
    remove gtest.m4 

Signed-off-by: Zhziyao <ziyao.zhang@intel.com>

uartie commented on Makefile.am:
```diff
@@ -6,7 +6,12 @@ DISTCLEANFILES = \
 	configure config.h.in config.h.in~ depcomp install-sh ltmain.sh \
 	Makefile.in missing
 
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on capi/Makefile.am:
```diff
@@ -34,7 +34,7 @@ libyami_capi_la_SOURCES     = $(libyami_capi_source_c)
 libyami_capi_la_CPPFLAGS    = $(libyami_capi_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on codecparsers/Makefile.am:
```diff
@@ -129,7 +129,7 @@ libyami_codecparser_la_SOURCES     = $(libyami_codecparser_source_c)
 libyami_codecparser_la_CPPFLAGS    = $(libyami_codecparser_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on common/Makefile.am:
```diff
@@ -50,7 +50,7 @@ libyami_common_la_SOURCES     = $(libyami_common_source_c)
 libyami_common_la_CPPFLAGS    = $(libyami_common_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on configure.ac:
```diff
@@ -382,6 +378,13 @@ fi
+AC_ARG_ENABLE([gtest],
+    [AC_HELP_STRING([--enable-gtest],
+                    [build unit tests @<:@default=no@:>@])],
+    [], [enable_gtest="no"])
```
Please use **--enable-tests** instead since it's possible we could switch test frameworks at some point in the future.  This also makes it more consistent with libva-utils and intel-vaapi-driver "--enable-tests" option.

uartie commented on configure.ac:
```diff
@@ -382,6 +378,13 @@ fi
+                    [build unit tests @<:@default=no@:>@])],
+    [], [enable_gtest="no"])
+
+AM_CONDITIONAL(ENABLE_GTEST, test "$enable_gtest" = "yes")
```
use ENABLE_TESTS instead

uartie commented on decoder/Makefile.am:
```diff
@@ -117,7 +117,7 @@ libyami_decoder_la_SOURCES     = $(libyami_decoder_source_c)
 libyami_decoder_la_CPPFLAGS    = $(libyami_decoder_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on encoder/Makefile.am:
```diff
@@ -93,7 +93,7 @@ libyami_encoder_la_SOURCES     = $(libyami_encoder_source_c)
 libyami_encoder_la_CPPFLAGS    = $(libyami_encoder_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on vpp/Makefile.am:
```diff
@@ -90,7 +90,7 @@ libyami_vpp_la_SOURCES      = $(libyami_vpp_source_c)
 libyami_vpp_la_CPPFLAGS     = $(libyami_vpp_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on vaapi/Makefile.am:
```diff
@@ -46,7 +46,7 @@ libyami_vaapi_la_SOURCES     = $(libyami_vaapi_source_c)
 libyami_vaapi_la_CPPFLAGS    = $(libyami_vaapi_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on v4l2/Makefile.am:
```diff
@@ -58,7 +58,7 @@ libyamiv4l2_la_SOURCES     = $(libyamiv4l2_source_c)
 libyamiv4l2_la_CPPFLAGS    = $(libyamiv4l2_cppflags) $(AM_CPPFLAGS)
 
-if ENABLE_UNITTESTS
+if ENABLE_GTEST
```
please use ENABLE_TESTS instead

uartie commented on capi/Makefile.unittest:
```diff
@@ -11,18 +11,18 @@ unittest_LDFLAGS = \
 unittest_CXXFLAGS = \
-	$(GTEST_CXXFLAGS) \
 	$(AM_CXXFLAGS) \
+	-pthread \
```
"-pthread" should be in LDFLAGS, not CXXFLAGS.  This applies to all the makefiles.

Zhziyao commented:
Hi, @uartie 
updated. please help to review. thanks.

uartie commented on capi/Makefile.am:
```diff
@@ -34,7 +34,7 @@ libyami_capi_la_SOURCES     = $(libyami_capi_source_c)
 
-if ENABLE_UNITTESTS
+if ENABLE_TESTS
 -include Makefile.unittest
```
typo

Zhziyao replied:
what's the typo? the capi dir doesn't contain any unittest case.

uartie replied:
Even though there are no test cases yet for capi, we should not disable it.  When we add test cases in the future, we won't have to remember to enable it.

uartie reviewed (changes requested):
One typo in capi/Makefile.am.

Also, please split this into two commits.  First commit should add gtest source and make note of the gtest version in commit message (see https://github.com/01org/intel-vaapi-driver/commit/1746adbb9adabd9fbee417ae3db67da599cc84e9 for example).  Second commit should modify unittest files to use the new gtest source and remove gtest.m4.

uartie commented:
Also, please rebase.

Zhziyao committed:
test: add googletest release-1.8.0 source
We will use the Google Test Framework for testing.

Google Test is not meant to be packaged and used as a system
dependency like other libraries.  Google Test recommends that
it be custom compiled for each project that uses it.  Thus,
we'll keep our own copy of it here and compile it as needed.

Signed-off-by: Zhziyao <ziyao.zhang@intel.com>

Zhziyao committed:
gtest:build and link to subdir gtest library

modify subdirs' unittest files
remove gtest.m4
replace --enable-gests=... configure option with --enable-tests

Signed-off-by: Zhziyao <ziyao.zhang@intel.com>

Zhziyao commented:
Hi, @uartie 
updated. Please help review.Thanks.

Zhziyao commented:
Hi, @uartie 
Thank you for your patient explanation.
updated and help review.

uartie reviewed (approved):
LGTM.

Please ignore the fedora 26 build failure.  It is unrelated to this pull-request.

uartie reviewed (changes requested):
Sorry I didn't notice this sooner... but, could you update the "Testing" section in the README to reflect these changes, too?

xuguangxin commented:
@uartie good suggestion, we did not notice it.
thanks

Zhziyao commented:
OK. The "Testing" section of README is modified. Please help review.

uartie reviewed (approved):
LGTM

xuguangxin commented:
thanks @uartie  for review
